src/utils/runner.ts

```diff
@@ -3,53 +3,120 @@ import { TestRunRequest, TestResult } from "../models/types";
 import { cloneRepository, cleanupRepository } from "./gitUtils";
 import { reportResults } from "./resultReporter";
 import { buildDockerImage, runInDocker, cleanupDocker } from "./dockerUtils";
-// import fs from "fs/promises";
-// import path from "path";
+import { loadTestFile } from "./testLoader";
+import { fetchUserProgress } from "./progress";
 import { EVENT_TYPE } from "./constants";
 import logger from "./logger";
+import fs from "fs/promises";
+import path from "path";
+import { ProgressResponse } from "../models/types";
 
 export async function runTestProcess(request: TestRunRequest): Promise<void> {
   const { repoUrl, branch, commitSha } = request;
+
   let repoDir: string | null = null;
   let imageName: string | null = null;
 
   try {
+    // Fetch user's progress
+    const progressData: ProgressResponse[] = await fetchUserProgress(repoUrl);
+    if (progressData.length === 0) {
+      throw new Error("No progress data found for the repository.");
+    }
+    const progress = progressData[0];
+    console.log("Progress", progress);
```

Comment:
Given that you're mostly using the logger instance, we can remove console.log and stick with the logger. Logging the progress here is a duplicate since you're also logging it in the next line
```suggestion
```

Reply:
Yea, this is just for debugging